kinto_http/__init__.py

```diff
@@ -118,6 +119,12 @@ def _paginated(self, endpoint, records=None, if_none_match=None, **kwargs):
 
         record_resp, headers = self.session.request(
             'get', endpoint, headers=headers, params=kwargs)
+
+        # Save the current records collection timestamp
+        if 'etag' in map(str.lower, headers.keys()):
+            etag = headers['ETag'].strip('"')
+            self._records_collection_timestamp[endpoint] = etag
```

Comment:
I think ETag is always present, but you could just do `headers.get('ETag', '')` no ?